src/rss_etl.py

```diff
@@ -64,39 +108,49 @@ def extract_metadata_from_rss_feed(self, max_attempts=3):
             if bill is None:
                 continue
 
-            bill_no, printer_no = self.get_bill_and_printer_no(bill.title)
-
-            self.create_and_attempt_to_insert_bill(bill, bill_no)
-            self.create_and_attempt_to_insert_revision(bill, bill_no, printer_no)
+            bill_identifier = BillIdentifier(bill["guid"])
+            self.create_and_attempt_to_insert_bill(bill, bill_identifier)
+            self.create_and_attempt_to_insert_revision(bill, bill_identifier)
             attempt_count += 1
             if attempt_count >= max_attempts:
                 break
 
-    def create_and_attempt_to_insert_revision(self, bill, bill_no, printer_no):
-        revisions_output_record = self.create_revisions_output_record(bill, bill_no, printer_no)
-        if not revisions_output_record.has_existing_supabase_records(["bill_id", "printer_no", "legislative_body"]):
+    def create_and_attempt_to_insert_revision(self, bill, bill_identifier: BillIdentifier):
+        revisions_output_record = self.create_revisions_output_record(bill, bill_identifier)
+        if not revisions_output_record.has_existing_supabase_records(["revision_guid"]):
             self.insert_new_record(revisions_output_record)
 
 
-    def create_and_attempt_to_insert_bill(self, bill, bill_no):
-        bills_output_record = self.create_bill_output_record(bill, bill_no)
-        if not bills_output_record.has_existing_supabase_records(["bill_id"]):
+    def create_and_attempt_to_insert_bill(self, bill, bill_identifier: BillIdentifier):
+        bills_output_record = self.create_bill_output_record(bill, bill_identifier)
+        if not bills_output_record.has_existing_supabase_records(["legislative_id"]):
             self.insert_new_record(bills_output_record)
 
-    def create_revisions_output_record(self, bill, bill_no, printer_no):
+    # Method to obtain bill internal id from existing supabase row for a bill
+    def get_bill_internal_id(self, bill_identifier: BillIdentifier):
+        statement = self.supa_con.table("Bills").select("bill_internal_id").eq("legislative_id", bill_identifier.bill_guid)
+        bill_internal_id = statement.execute()
+        return bill_internal_id
+
+    def create_revisions_output_record(self, bill, bill_identifier: BillIdentifier):
         revisions_output_record = OutputRecord(self.supa_con, "Revisions", REVISIONS_OUTPUT_DICT)
-        revisions_output_record.output_dict["bill_id"] = bill_no
-        revisions_output_record.output_dict["printer_no"] = printer_no
+        revisions_output_record.output_dict["bill_internal_id"] = self.get_bill_internal_id(bill_identifier)
```

Comment:
We need to finalize our choices for what the revisions/bills table primary key will be. Are we using the GUID provided from the RSS feed, or a randomly generated one? What is the relationship between `bill_id` and bill_internal_id`. If you want to make it clear, though  this will require discussion between mutliple people, "i recommend calling it `bill_rss_guid` or `bill_gov_guid` if using the rss one, or if generating it, call it `bill_generated_id`, or `bill_postgres_id` or `bill_primary_key`, etc.

Reply:
The code you're referencing is outdated -- in the new version, I've removed the "bill_id" and made the column nomenclature more clear. 

I also agree that we should use the GUID as the primary key. The issue is whether we should have a numeric internal ID for linking, or if the GUID is sufficient. 

But more prominently, to what degree do we want to block moving forward on this because of that? If we decide we want to get rid of the bill_internal_id, that would be a relatively trivial change in the database and these scripts -- we'd simply substitute them for the guid.